engine/src/flutter/impeller/display_list/canvas.cc

```diff
@@ -1467,6 +1467,50 @@ bool Canvas::Restore() {
   return true;
 }
 
+bool Canvas::AttemptBlurredTextOptimization(
+    const std::shared_ptr<TextFrame>& text_frame,
+    const std::shared_ptr<TextContents>& text_contents,
+    Entity& entity,
+    const Paint& paint) {
+  if (!paint.mask_blur_descriptor.has_value() ||  //
+      paint.image_filter != nullptr ||            //
+      paint.color_filter != nullptr ||            //
+      paint.invert_colors) {
+    return false;
+  }
+
+  // TODO(bdero): This mask blur application is a hack. It will always wind up
+  //              doing a gaussian blur that affects the color source itself
+  //              instead of just the mask. The color filter text support
+  //              needs to be reworked in order to interact correctly with
+  //              mask filters.
+  //              https://github.com/flutter/flutter/issues/133297
+  std::shared_ptr<FilterContents> filter =
+      paint.mask_blur_descriptor->CreateMaskBlur(
+          FilterInput::Make(text_contents),
+          /*is_solid_color=*/true, GetCurrentTransform());
+
+  std::optional<Glyph> maybe_glyph = text_frame->AsSingleGlyph();
+  int64_t identifier = maybe_glyph.has_value()
+                           ? maybe_glyph.value().index
+                           : reinterpret_cast<int64_t>(text_frame.get());
```

Comment:
question for @gaaclarke : is this safe/reasonable? The goal is only caching the same text frame. An alternative approach would be to have an ID system somewhere

Reply:
I think so.  Worst case scenario is that the text_frame gets copied and you get a cache miss.  Since we have a std::shared_ptr to it I don't expect we'll copy it (but I wouldn't rule it out).

Doesn't the TextFrame have a font itself too?  It's weird that we hash on the identifier which represents a font and have the font separately.  Can the TextFrame be mutated?

Reply:
Yeah hashing the font is for the "single glyph" optimization for Icons, but its unnecessary for the general text frame case.